src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java

```diff
@@ -0,0 +1,107 @@
+
+        when(definitionRepository.getBaseTypes()).thenReturn(Collections.emptyList());
+        BaseType.setCaseDefinitionRepository(definitionRepository);
+        BaseType.initialise();
+
+        when(fixedListBaseType.getType()).thenReturn(FixedRadioListValidator.TYPE_ID);
+        when(fixedListBaseType.getRegularExpression()).thenReturn(null);
+        BaseType.register(fixedListBaseType);
+
+        validator = new FixedRadioListValidator();
+
+        caseField = caseField().build();
+    }
+
+    @Test
+    void validValue() {
+        final List<ValidationResult> result01 = validator.validate("TEST_FIELD_ID",
+                                                                   NODE_FACTORY.textNode("AAAAAA"),
+                                                                   caseField);
+        assertEquals(0, result01.size());
+    }
+
+    @Test
+    void invalidValue() {
+        final List<ValidationResult> result01 = validator.validate("TEST_FIELD_ID",
+                                                                   NODE_FACTORY.textNode("DDDD"),
+                                                                   caseField);
+        assertEquals(1, result01.size(), result01.toString());
+    }
+
+    @Test
+    void nullValue() {
+        assertEquals(0, validator.validate("TEST_FIELD_ID", null, null).size(), "Did not catch NULL");
+    }
+
+    @Test
+    void getType() {
+        assertEquals(validator.getType(), BaseType.get("FixedRadioList"), "Type is incorrect");
+    }
+
+    @Test
+    void fieldTypeRegEx() {
+        final CaseField caseFieldWithRegEx = caseField().withRegExp("AAAAAA").build();
+        final List<ValidationResult> result01 = validator.validate("TEST_FIELD_ID", NODE_FACTORY.textNode("AAAAAA"),
+                                                                   caseFieldWithRegEx);
+        assertEquals(0, result01.size());
+
+        final List<ValidationResult> result02 = validator.validate("TEST_FIELD_ID", NODE_FACTORY.textNode("BBBBBB"),
+                                                                   caseFieldWithRegEx);
+        assertEquals(1, result02.size(), "BBBBBB failed regular expression check");
+        assertEquals(REGEX_GUIDANCE, result02.get(0).getErrorMessage());
+        assertEquals("TEST_FIELD_ID", result02.get(0).getFieldId());
+    }
+
+    @Test
+    void baseTypeRegEx() {
+        when(fixedListBaseType.getRegularExpression()).thenReturn("InvalidRegEx");
+        final List<ValidationResult> result = validator.validate("TEST_FIELD_ID",
+                                                                 NODE_FACTORY.textNode("AA"), caseField);
+        assertEquals(1, result.size(), "RegEx validation failed");
+        assertEquals("'AA' failed FixedList Type Regex check: InvalidRegEx", result.get(0).getErrorMessage());
+        assertEquals("TEST_FIELD_ID", result.get(0).getFieldId());
+    }
+
+    private CaseFieldBuilder caseField() {
```

Comment:
Would it be better to use TestBuildersUtil.CaseFieldBuilder for code reuse? May be merge `CaseFieldBuilder` class with the one in the TestBuildersUtil class?

Reply:
I have now created a tech debt ticket on this https://tools.hmcts.net/jira/browse/RDM-3366